File: template-method/c-sharp/DataAnalyzerBase.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TemplateMethod
{
    // Metot iskeletimizi ve soyutlanmış metotları içermektedir.
    // UML diyagramındaki AbstractClass yapısına denk gelmektedir.
    abstract class DataAnalyzerBase
    {
        // Metot iskeletimiz.
        public void AnalyzeDataOfLastWeek()
        {
            object data = GetDataOfLastWeek();
            object result = AnalyzeData(data);
            ExportResult(result);
        }

        private object GetDataOfLastWeek()
        {
            // Burada veri verinin çağrılma süreçleri yer alacaktır.
            // Default olarak bir obje dönülmektedir.
            Console.WriteLine("Method worked: GetDataOfLastWeek");
            return new object();
        }

        private object AnalyzeData(object data)
        {
            // Burada verinin analizi süreci yer almaktadır.
            // Analiz sonuçları default olarak bir obje dönmektedir.
            Console.WriteLine("Method worked: AnalyzeData");
            return new object();
        }

        // Alt sınıflar tarafından uygulanmasını istediğimiz metottur.
        // Metot iskeleti içerisinde farklılık gösterecek olan metotumuzdur.
        public abstract void ExportResult(object result);
    }
}

File: template-method/go/data-analyzer-base.go
package main

import "fmt"

// Metot iskeletimizi ve soyutlanmış metotları içermektedir.
// UML diyagramındaki AbstractClass yapısına denk gelmektedir.

type DataAnalyzer interface {
	exportResult([]byte)
}

type DataAnalyzerBase struct {
	dataAnalyzer DataAnalyzer
}

func NewDataAnalyzerBase() *DataAnalyzerBase {
	return &DataAnalyzerBase{}
}

func (d DataAnalyzerBase) AnalyzeDataOfLastWeek() {
	data := d.getDataOfLastWeek()
	result := d.analyzeData(data)
	d.dataAnalyzer.exportResult(result)
}

func (d DataAnalyzerBase) getDataOfLastWeek() []byte {
	// Burada veri verinin çağrılma süreçleri yer alacaktır.
	// Default olarak bir obje dönülmektedir.
	fmt.Println("Method worked: GetDataOfLastWeek")
	return []byte("data")
}

func (d DataAnalyzerBase) analyzeData(data []byte) []byte {
	// Burada verinin analizi süreci yer almaktadır.
	// Analiz sonuçları default olarak bir obje dönmektedir.
	fmt.Println("Method worked: AnalyzeData")
	return []byte("result")
}
